Simpletest

daniel-dgi opened the pull request.

daniel-dgi committed:
Renamed a few hooks to use islandora_xacml_api instead of islandora_xacml_editor.  Started working on tests for the api module.

daniel-dgi committed:
Added a web test for the api, unit tests for the Xacml class, and fixed up some documentation strings

daniel-dgi committed:
Added tests for removing users and adding/removing roles.

daniel-dgi committed:
Committing before removing a huge chunk of what i feel is probably unneccessary test code.

daniel-dgi committed:
Removed a bunch of test code that inspected the results of the Xacml class's default constructor with no arguments

daniel-dgi committed:
Added tests for isPopulated() and hasPermission().  Removed similar tests from the functions that check the DOM's

daniel-dgi committed:
Getting rid of old web test case for the api's module.  Can always add it back later.

jonathangreen commented:
I think the tests look awesome so far. Can we add a couple more tests? 
- Test that the correct methods are added for each rule type:
  - management rule
  - viewing rule
  - datastream rule
- Test that when you add a mimetype it ends up in the dom
- Test when you add a regular expression it ends up in the dom
- Test when you add a DSID it ends up in the dom

daniel-dgi committed:
Added tests to ensure the expected methods are in the management, viewing, and datastream rules.

daniel-dgi committed:
Added test to assert Mimetypes added to a DatastreamRule are present in the Xacml's DOM.

daniel-dgi committed:
Added unit test to assert added DSID's appear in the Xacml's DOM.

daniel-dgi committed:
Added tests to assert that added mimetype regex's appear in the Xacml's DOM.

daniel-dgi committed:
Changes to make the islandora_xacml_api module conform to style guides.

daniel-dgi committed:
Fixed some typos in comments.

daniel-dgi commented:
Take a look.  I added the tests you requested, ran the module through Coder, and cleaned it up so that it passes.  Let me know if there's anything else you'd like to test.

On 2012-11-14, at 12:22 PM, Jonathan Green notifications@github.com wrote:

> I think the tests look awesome so far. Can we add a couple more tests?
> 
> Test that the correct methods are added for each rule type:
> 
> management rule
> viewing rule
> datastream rule
> Test that when you add a mimetype it ends up in the dom
> 
> Test when you add a regular expression it ends up in the dom
> 
> Test when you add a DSID it ends up in the dom
> 
> —
> Reply to this email directly or view it on GitHub.

daniel-dgi committed:
Forgot a comment header above a test function.
